include/VertexBookkeeping.h

```diff
@@ -27,20 +28,21 @@ struct VertexBookkeeping {
 
 public:
     VertexBookkeeping()
-            : VertexBookkeeping(0, 0, 0)
+            : VertexBookkeeping(0, 0, 0, 0)
     {}
 
-    VertexBookkeeping(PreferenceList::SizeType begin, PreferenceList::SizeType end,
+    VertexBookkeeping(PreferenceList::SizeType begin, PreferenceList::SizeType end, PreferenceList::SizeType tied_index,
```

Comment:
We still need a VertexBookkeeping constructor which takes a begin, and residual arguments.
This should be a new constructor.